src/components/workAllocation/WorkAllocation.tsx

```diff
@@ -309,18 +316,51 @@ export default function WorkAllocation() {
                     })}
                   />
                 </div>
-
-                <div className="md:flex-grow">
-                  <CustomReactSelect
-                    label="Project (cost code description)"
-                    name="project"
-                    dataTestId="project"
-                    fixedWidth={210}
-                    emptyOption={true}
-                    options={selectOptionValues(projects, 'name', 'name', true, { sort: true, alphaFirst: true })}
+                {addNewProjectCodeCode ? (
+                  <AddNewConfigOption
+                    inputRef={addNewConfigOptionInputRef}
+                    returnedDataObject="addProject"
+                    onSubmit={(value: string) => {
+                      return stanCore.AddProject({ name: value });
+                    }}
+                    onCancel={() => {
+                      setAddNewProjectCodeCode(false);
+                    }}
+                    onSuccess={() => {
+                      send({
+                        type: 'ADD_NEWLY_CREATED_PROJECT',
+                        project: projectFactory.build({ name: addNewConfigOptionInputRef.current!.value })
+                      });
+                      setFieldValue('project', addNewConfigOptionInputRef.current!.value);
+                    }}
+                    onFinish={() => {
+                      setAddNewProjectCodeCode(false);
+                    }}
+                    configLabel="Add New Project"
+                    configName="project"
                   />
-                </div>
-
+                ) : (
+                  <div className="flex">
+                    <CustomReactSelect
+                      className="w-6/7"
+                      label="Project (cost code description)"
+                      name="project"
+                      dataTestId="project"
+                      emptyOption={true}
+                      options={selectOptionValues(projects, 'name', 'name', true, { sort: true, alphaFirst: true })}
+                      value={values.project}
+                    />
+                    <AddButton
+                      data-testid="addNewProject-btn"
+                      className={errors.project ?? 'place-self-end'}
+                      onClick={(event) => {
+                        event.preventDefault();
+                        event.stopPropagation();
+                        setAddNewProjectCodeCode(true);
+                      }}
```

Comment:
Cost code add button is slightly mis-aligned (vertically) in UI
![Screenshot 2024-02-12 at 12 46 09](https://github.com/sanger/stan-client/assets/55585488/e0b0c4f1-2a6c-484d-9163-009cb480a331)